bin/reth/src/count_opcodes/mod.rs

```diff
@@ -0,0 +1,432 @@
+            Bytes::from_iter(test_bytecode_bytes.iter().filter(|op| **op != 0xaa));
+        assert_eq!(manually_filtered_bytes, filter_bytecode_bytes(&test_bytecode_bytes))
+    }
+
+    #[test]
+    fn opcode_counter() {
+        let test_bytecode = all_opcodes_test_string();
+        let bytecode = Bytecode::new_raw(hex::decode(&test_bytecode).unwrap().into());
+        let filtered_bytes = filter_bytecode_bytes(bytecode.bytes());
+
+        let mut opcode_counter = OpCodeCounter::new();
+        opcode_counter.count_sequences(&filtered_bytes);
+
+        // check that every opcode is in our opcodes map
+        for (i, opcode) in OPCODE_JUMPMAP.iter().enumerate() {
+            if opcode.is_some() {
+                let i = i as u8;
+                assert_eq!(opcode_counter.opcodes.get(&i), Some(1_usize).as_ref());
+            }
+        }
+    }
+
+    #[test]
+    fn opcode_counter_tuples() {
+        let test_bytecode = opcodes_tuples_test_string();
+        let bytecode = Bytecode::new_raw(hex::decode(test_bytecode).unwrap().into());
+        let filtered_bytes = filter_bytecode_bytes(bytecode.bytes());
+
+        let mut opcode_counter = OpCodeCounter::new();
+        opcode_counter.count_sequences(&filtered_bytes);
+
+        assert_eq!(opcode_counter.opcodes.get(&0).unwrap().clone(), 1);
+        assert_eq!(opcode_counter.tuple_opcodes.get(&[1, 2]).unwrap().clone(), 2);
+        assert_eq!(opcode_counter.triplets_opcodes.get(&[3, 4, 5]).unwrap().clone(), 2);
+        assert_eq!(opcode_counter.quadruplets_opcodes.get(&[6, 7, 8, 9]).unwrap().clone(), 2);
+    }
+
+    fn opcodes_tuples_test_string() -> String {
+        let test_bytecode = "00"; // STOP
+
+        let test_bytecode = format!("{test_bytecode}01"); // ADD
+        let test_bytecode = format!("{test_bytecode}02"); // MUL
+        let test_bytecode = format!("{test_bytecode}01"); // ADD
+        let test_bytecode = format!("{test_bytecode}02"); // MUL
+
+        let test_bytecode = format!("{test_bytecode}03"); // SUB
+        let test_bytecode = format!("{test_bytecode}04"); // DIV
+        let test_bytecode = format!("{test_bytecode}05"); // SDIV
+        let test_bytecode = format!("{test_bytecode}03"); // SUB
+        let test_bytecode = format!("{test_bytecode}04"); // DIV
+        let test_bytecode = format!("{test_bytecode}05"); // SDIV
+
+        let test_bytecode = format!("{test_bytecode}06"); // MOD
+        let test_bytecode = format!("{test_bytecode}07"); // SMOD
+        let test_bytecode = format!("{test_bytecode}08"); // ADDMOD
+        let test_bytecode = format!("{test_bytecode}09"); // MULMOD
+        let test_bytecode = format!("{test_bytecode}06"); // MOD
+        let test_bytecode = format!("{test_bytecode}07"); // SMOD
+        let test_bytecode = format!("{test_bytecode}08"); // ADDMOD
+        let test_bytecode = format!("{test_bytecode}09"); // MULMOD
+
+        test_bytecode
+    }
+
+    fn all_opcodes_test_string() -> String {
```

Comment:
This can be quite noisy in the file, however we wanted to make it as easy as possible to read and modify this test opcodes string if needed